core-blocks/latest-comments/edit.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	Disabled,
+	PanelBody,
+	RangeControl,
+	ToggleControl,
+	ServerSideRender,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	InspectorControls,
+	BlockAlignmentToolbar,
+	BlockControls,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies.
+ */
+import './editor.scss';
+
+const MIN_COMMENTS = 1;
+const MAX_COMMENTS = 100;
+
+class LatestComments extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setAlignment = this.setAlignment.bind( this );
+		this.setCommentsToShow = this.setCommentsToShow.bind( this );
+		this.toggleAttribute = this.toggleAttribute.bind( this );
+	}
+
+	toggleAttribute( propName ) {
+		return () => {
+			const value = this.props.attributes[ propName ];
+			const { setAttributes } = this.props;
+
+			setAttributes( { [ propName ]: ! value } );
+		};
+	}
+
+	setAlignment( align ) {
+		this.props.setAttributes( { align } );
+	}
+
+	setCommentsToShow( commentsToShow ) {
+		this.props.setAttributes( { commentsToShow } );
+	}
+
+	render() {
+		const {
+			align,
+			commentsToShow,
+			displayAvatar,
+			displayDate,
+			displayExcerpt,
+		} = this.props.attributes;
+
+		return (
+			<Fragment>
+				<BlockControls>
+					<BlockAlignmentToolbar
+						value={ align }
+						onChange={ this.setAlignment }
+					/>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Latest Comments Settings' ) }>
+						<ToggleControl
+							label={ __( 'Display avatar' ) }
+							checked={ displayAvatar }
+							onChange={ this.toggleAttribute( 'displayAvatar' ) }
```

Comment:
`toggleAttribute` returns a new function every invocation, causing React to do render reconciliation more often than we need it to. I wonder if we should pre-bind these functions in the constructor:

```js
constructor() {
	// ...

	this.toggleDisplayAvatar = this.createToggleAttribute( 'displayAvatar' );
}
```

Reply:
Yeah, this has been pointed out elsewhere that this pattern is bad for unneeded re-renders, though obviously the code is nice.

Pre-binding them in the constructor is a good pattern actually 👍